roles/grafana/tasks/main.yml

```diff
@@ -0,0 +1,79 @@
+---
+- name: create config dir
+  file:
+    path: "{{ grafana_config_dir }}"
+    state: directory
+
+- name: create data dir
+  file:
+    path: "{{ grafana_data_dir }}"
+    state: directory
+    owner: "{{ grafana_user_id }}"
+
+- name: create dashboard dir
+  file:
+    path: "{{ grafana_cfg_dashboard_path }}"
+    state: directory
+    owner: "{{ grafana_user_id }}"
+
+- name: create provisioning dirs
+  file:
+    path: "{{ grafana_config_dir }}/provisioning/{{ item }}"
+    state: directory
+    owner: "{{ grafana_user_id }}"
+  with_items:
+    - datasources
+    - dashboards
+
+- name: copy dashboards
+  copy:
+    src: "{{ item }}"
+    dest: "{{ grafana_cfg_dashboard_path }}"
+    owner: "{{ grafana_user_id }}"
+  with_items:
+    - gpu-dashboard.json
+  notify: restart grafana
+
+- name: install datasource config
+  template:
+    src: "templates/prometheus-datasource.yml.j2"
+    dest: "{{ grafana_config_dir }}/provisioning/datasources/prometheus-datasource.yml"
+    owner: root
+    group: root
+    mode: 0644
+  notify: restart grafana
+
+- name: install dashboard config
+  template:
+    src: "templates/prometheus-dashboard.yml.j2"
+    dest: "{{ grafana_config_dir }}/provisioning/dashboards/prometheus-dashboard.yml"
+    owner: root
+    group: root
+    mode: 0644
+  notify: restart grafana
+
+- name: install config file
+  template:
+    src: templates/grafana.ini.j2
+    dest: "{{ grafana_config_dir }}/grafana.ini"
+    owner: root
+    group: root
+    mode: 0644
+  notify: restart grafana
+
+- name: install systemd unit file
+  template:
+    src: templates/docker.grafana.service.j2
+    dest: "/etc/systemd/system/{{ grafana_svc_name }}"
```

Comment:
Consider this path instead? `/usr/local/lib/systemd/system/`

I think of `/etc/systemd/system/` as primarily for overrides and links.

Reply:
My understanding has always been that best-practice is to use `/etc/systemd/system`. I also prefer to keep configuration confined to `/etc` because it's easier to back up.

https://unix.stackexchange.com/questions/224992/where-do-i-put-my-systemd-unit-file

Reply:
Nice link. I'm convinced.